core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -695,7 +695,8 @@ private[spark] class TaskSetManager(
   def handleSuccessfulTask(tid: Long, result: DirectTaskResult[_]): Unit = {
     val info = taskInfos(tid)
     val index = info.index
-    info.markFinished(TaskState.FINISHED)
+    // Mark task as finished. Note that finishing time here should be bigger than 0.
```

Comment:
Thanks for review :) what I want to say is that `finishTime` of `TaskInfo` should be set larger than 0, otherwise `TaskInfo.finishTime` will return false. I removed the comments and added an `assert` inside `TaskInfo`.